tests/rptest/tests/acls_test.py

```diff
@@ -357,6 +356,88 @@ def test_mtls_principal(self, rules=None, fail=False):
+            pass_w_base_user=False,
+            pass_w_cluster_user=True,
+            pass_w_super_user=True,
+            err_msg='check_permissions failed before migration')
+
+        # Change the authn method and initiate feature migration
+        # with a rolling restart. Do not recreate tls certs.
+        self.security.endpoint_authn_method = authn_method
+        self.redpanda.set_security_settings(self.security)
+        self.logger.debug('Initiating rolling restart')
+        self.redpanda.rolling_restart_nodes(self.redpanda.nodes,
+                                            start_timeout=90,
+                                            stop_timeout=90)
+
+        # Check that the authn_method was set on all brokers
+        pattern = 'Started Kafka API server.*:{' + authn_method + '}.*'
+        assert self.redpanda.search_log_all(pattern)
+
+        # Once restart is complete, check permissions should succeed when authn_method
+        # is sasl because the system will validate against SASL/SCRAM creds already stored
+        # on the brokers. However, when authn_method is mtls_identiy, check permissions
+        # should fail because no principal mapping rule is set yet.
+        if authn_method == 'sasl':
+            self.check_permissions(
+                pass_w_base_user=False,
+                pass_w_cluster_user=True,
+                pass_w_super_user=True,
+                err_msg='check_permissions failed in intermediate state')
+        elif authn_method == 'mtls_identity':
+            try:
+                self.check_permissions(pass_w_base_user=False,
+                                       pass_w_cluster_user=True,
+                                       pass_w_super_user=True,
+                                       timeout_sec=10)
+                raise RuntimeError(
+                    'check_perms should have failed in intermediate state')
+            except TimeoutError:
+                # Expect a timeout failure which comes from
+                # wait_until. The timeout is because check_perms
+                # should fail.
+                pass
+
+        # Change cluster wide configs for kafka_enable_authorization and
+        # kafka_mtls_principal_mapping_rules via the admin api.
+        # Do not expect a restart.
+        #
+        # NOTE:
+        # - kafka_enable_authorization overrides sasl in an authorization context
+        # - authentication_method overrides sasl in an authentication context
+        new_cfg = {'kafka_enable_authorization': True}
+        if authn_method == 'mtls_identity':
+            new_cfg['kafka_mtls_principal_mapping_rules'] = [
+                'RULE:.*CN=([^,]+).*/$1/', 'DEFAULT'
+            ]
+        self.redpanda.set_cluster_config(new_cfg)
+
+        # Check that new cluster configs are set
+        cluster_cfg = self.admin.get_cluster_config()
+        assert cluster_cfg['kafka_enable_authorization'] == new_cfg[
+            'kafka_enable_authorization']
```

Comment:
nit: This check is done inside `set_cluster_config`

Reply:
I hear you, the alternative is to add a comment `# config check done automatically in set_cluster_config` but I think an explicit line of code is more robust so I'll leave it in.